### PR TITLE
Scripts/Scarlet Enclave: Death Comes From On High

### LINKED ISSUE #20843: [3.3.5][Quest] Death Comes From On High (12641)
**Description:**

The Eye of Acherus does not behave as it should in regards to launching and its movement speed.

**Current behaviour:**

At the start of the quest, the Eye is not launched towards New Avalon. The Movement speed of the Eye is much slower than retail.

**Expected behaviour:**

The Eye of Acherus should launch towards New Avalon with notifications for "The Eye of Acherus launches towards its destination." and "The Eye of Acherus is in your control."
After gaining control the Eye should also move faster. 
Video for reference: https://www.youtube.com/watch?v=VAQwxyWrXrc

**Steps to reproduce the problem:**

1. Create a new deathknight.
2. Complete the intro quest chain up until this point.
3. Begin quest and use the Eye of Acherus.

**Branch(es):**

3.3.5

**TC rev. hash/commit:** 

TrinityCore rev. 3e74ef1528db 2017-11-10 23:45:42 +0100 (3.3.5 branch) (Unix, RelWithDebInfo, Static)

**TDB version:** TDB_full_world_335.63_2017_04_18

**Operating system:** Fedora 26

### PR DESCRIPTION
**Pull request by DanVS**
**Changes proposed:**
Possible fix for quest [Death Comes From On High (12641)](https://woehead.way-of-elendil.fr/?quest=12641). Schedule launch event on initialization stage.

**Target branch(es):** 3.3.5

**Issues addressed:** Closes #20843

**Tests performed:** Does build and works in-game.

### CODE REVIEW

File: src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp [hunk 1]
```diff
@@ -455,6 +448,7 @@ class npc_eye_of_acherus : public CreatureScript
         }
 };
 
+
```
Comment (Aokromes) [inline src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp]: remove :)

Comment (DanVS) [reply src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp]: Fixed in less then 1 minute.

File: sql/updates/world/3.3.5/9999_99_99_99_world.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+-- Eye of Acherus (28511)
+UPDATE `creature_template` SET `modelid1`=26320, `modelid2`=0 WHERE `entry`=28511;
```
Comment (Aokromes) [inline sql/updates/world/3.3.5/9999_99_99_99_world.sql]: this is hack without sniffs.

Comment (DanVS) [reply sql/updates/world/3.3.5/9999_99_99_99_world.sql]: Restored previous hack.

File: src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp [hunk 3]
```diff
@@ -376,16 +376,18 @@ class npc_eye_of_acherus : public CreatureScript
         {
             npc_eye_of_acherusAI(Creature* creature) : ScriptedAI(creature)
             {
-                me->SetDisplayId(me->GetCreatureTemplate()->Modelid1);
-                if (Player* owner = me->GetCharmerOrOwner()->ToPlayer())
-                    owner->SendAutoRepeatCancel(me);
-
+            	me->SetDisplayId(me->GetCreatureTemplate()->Modelid1);
```
Comment (Aokromes) [inline src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp]: 1 space more :)

Comment (DanVS) [reply src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp]: Fixed :+1:

**Comment by illfated** [comment]
@DanVS : title suggestion: **Scripts/Scarlet Enclave: Death Comes From On High**
"Core" and "Scripts" are generally 2 separate main categories, like Core/Spells vs Scripts/Spells.
When it comes to C++ scripts, it is usually **Scripts/** + area name or dungeon name.

File: src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp [hunk 4]
```diff
@@ -434,14 +435,7 @@ class npc_eye_of_acherus : public CreatureScript
...
                     me->SetDisableGravity(false);
-                    DoCast(me, SPELL_EYE_FLIGHT);
-                }
-
-                if (movementType == POINT_MOTION_TYPE && pointId == POINT_EYE_FALL)
-                {
-                    me->SetDisableGravity(true);
-                    me->SetControlled(true, UNIT_STATE_ROOT);
-                    _events.ScheduleEvent(EVENT_MOVE_START, 5000);
+                    me->CastSpell(me, SPELL_EYE_FLIGHT);
```
Comment (Keader) [inline src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp]: DoCastSelf

File: src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp [hunk 5]
```diff
@@ -377,15 +377,17 @@ class npc_eye_of_acherus : public CreatureScript
             npc_eye_of_acherusAI(Creature* creature) : ScriptedAI(creature)
             {
                 me->SetDisplayId(me->GetCreatureTemplate()->Modelid1);
-                if (Player* owner = me->GetCharmerOrOwner()->ToPlayer())
-                    owner->SendAutoRepeatCancel(me);
-
+                me->CastSpell(me, SPELL_EYE_VISUAL, true);
```
Comment (Keader) [inline src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp]: DoCastSelf

File: src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp [hunk 6]
```diff
@@ -377,15 +377,17 @@ class npc_eye_of_acherus : public CreatureScript
...
                 me->SetReactState(REACT_PASSIVE);
+                me->SetDisableGravity(true);
+                me->SetControlled(true, UNIT_STATE_ROOT);
 
                 Movement::MoveSplineInit init(me);
                 init.MoveTo(EyeOFAcherusFallPoint.GetPositionX(), EyeOFAcherusFallPoint.GetPositionY(), EyeOFAcherusFallPoint.GetPositionZ(), false);
                 init.SetFall();
                 me->GetMotionMaster()->LaunchMoveSpline(std::move(init), POINT_EYE_FALL, MOTION_PRIORITY_NORMAL, POINT_MOTION_TYPE);
+
+                _events.ScheduleEvent(EVENT_MOVE_START, 7000);
```
Comment (Keader) [inline src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp]: _events.ScheduleEvent(EVENT_MOVE_START, 7s);

File: src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp [hunk 7]
```diff
@@ -400,9 +402,9 @@ class npc_eye_of_acherus : public CreatureScript
                     {
                         case EVENT_MOVE_START:
                         {
-                            DoCast(me, SPELL_EYE_FLIGHT_BOOST);
-
+                            me->CastSpell(me, SPELL_EYE_FLIGHT_BOOST);
```
Comment (Keader) [inline src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp]: DoCastSelf

File: src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp (hunk 4 above)
Comment (illfated) [reply src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp]: ... using the format
```cpp
DoCastSelf(SPELL_EYE_FLIGHT);
```
(if I recall correctly)

**Comment by Faq** [comment]
why this was closed?

**Pull request closed without merging**